### DISCUSSION TITLE
Default BeeWare icon shows in some places athough it was changed

### DISCUSSION THREAD

**Discussion by MarusGradinaru** [beeware/beeware#407] [General]
I have a problem. I created an Android app with BeeWare Briefcase tool and changed all ic_launcher default icons (the brutus bee) and when I install and run the app, the launcher icon is indeed changed, but it still persist in the Toast messages and in the notification messages (only the unexpanded version). I searched the entire project folder and that default icon does not exists anywhare. I don't understand this behavior... What can I do change that icon in that 2 places I mentioned ? I even uninstalled the app but it still persist...
![image](https://github.com/user-attachments/assets/c23d0288-2310-4346-ad29-959fc2734c12)
![image](https://github.com/user-attachments/assets/4ec0b0ef-51cc-45a1-a25f-55d322bfbaab)

**Comment by freakboy3742** [comment]
I can think of four possible causes here.

The first is that you haven't updated the app to add resources. App resources (like icons) aren't updated by default; you need to run `briefcase update --update-resources` or `briefcase run --update-resources` to get new icons copied into the app. If you're not getting *any* of the new icons, this is the most likely cause. When you run this command, you should see log messages indicating that the icons have been copied into place; if you don't... then they wont' be :-)

The second is that you've made a call with `--update-resources`, but you've specified the icons incorrectly. Again, the log messages are your friend here; you're looking for messages that looks something like:
```
[testbed] Installing application resources...
Installing icons/testbed-round-48.png as 48px round application icon... started
Installing icons/testbed-round-48.png as 48px round application icon... done
Installing icons/testbed-round-72.png as 72px round application icon... started
Installing icons/testbed-round-72.png as 72px round application icon... done
...
Unable to find icons/testbed-adaptive-162.png for 162px adaptive application icon; using default
Unable to find icons/testbed-adaptive-216.png for 216px adaptive application icon; using default
Unable to find icons/testbed-adaptive-324.png for 324px adaptive application icon; using default
Unable to find icons/testbed-adaptive-432.png for 432px adaptive application icon; using default
```
The former (started... done) indicates that the icon was found successfully; the latter (unable to find...) indicates the icon wasn't specified correctly.

Third, it could be a platform issue. Different versions (and vendors) of Android have various caches for icons; and sometimes old icons get "stuck" in the cache. I don't know any reliable way to flush these caches; the best approach I'm aware of is to manually delete the app from the phone, then re-add it.

Lastly, it could be a feature issue. Briefcase installs a set of icons, but Android is a platform that changes rapidly between versions, and has some inconsistencies between vendors, so we might be missing an icon size that your particular install needs. If a particular version/vendor combination is looking for an icon of size X, and it can't find it, it will fall back to a different size; and sometimes that fallback strategy can be complicated. The fact you're not getting *any* of your icons makes this the least likely, but I can't rule it out completely.

**Comment by MarusGradinaru** [comment]
It seems that was a system bug, because I restarted the phone and now the icons for Toast and Notification are shown correctly.
